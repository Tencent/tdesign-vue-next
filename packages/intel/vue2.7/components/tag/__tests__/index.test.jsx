import { mount } from '@vue/test-utils';
import Tag from '@/src/tag/index.ts';
import CheckTag from '@/src/tag/check-tag.tsx';

describe('Tag or CheckTag', () => {
  describe(':Tag:props', () => {
    it(':theme:', () => {
      const wrapper = mount({
        render() {
          return <Tag></Tag>;
        },
      });
      expect(wrapper.element).toMatchSnapshot();
    });
    it(':theme:default', () => {
      const wrapper = mount({
        render() {
          return <Tag theme={'default'}></Tag>;
        },
      });
      expect(wrapper.element).toMatchSnapshot();
    });
    it(':theme:primary', () => {
      const wrapper = mount({
        render() {
          return <Tag theme={'primary'}></Tag>;
        },
      });
      expect(wrapper.element).toMatchSnapshot();
    });
    it(':theme:warning', () => {
      const wrapper = mount({
        render() {
          return <Tag theme={'warning'}></Tag>;
        },
      });
      expect(wrapper.element).toMatchSnapshot();
    });
    it(':theme:danger', () => {
      const wrapper = mount({
        render() {
          return <Tag theme={'danger'}></Tag>;
        },
      });
      expect(wrapper.element).toMatchSnapshot();
    });
    it(':theme:success', () => {
      const wrapper = mount({
        render() {
          return <Tag theme={'success'}></Tag>;
        },
      });
      expect(wrapper.element).toMatchSnapshot();
    });

    it(':size:small', () => {
      const wrapper = mount({
        render() {
          <Tag size="small"></Tag>;
        },
      });
      expect(wrapper.element).toMatchSnapshot();
    });
    it(':size:medium', () => {
      const wrapper = mount({
        render() {
          <Tag size="medium"></Tag>;
        },
      });
      expect(wrapper.element).toMatchSnapshot();
    });
    it(':size:large', () => {
      const wrapper = mount({
        render() {
          return <Tag size={'large'}></Tag>;
        },
      });
      expect(wrapper.element).toMatchSnapshot();
    });

    it(':closable', () => {
      const fn = vi.fn();
      const wrapper = mount({
        render() {
          return <Tag closable onClose={fn}></Tag>;
        },
      });
      expect(wrapper.element).toMatchSnapshot();
      wrapper.find('.t-icon-close').trigger('click');
      expect(fn).toHaveBeenCalled();
    });

    it(':checked', () => {
      const wrapper = mount({
        render() {
          return <Tag checked></Tag>;
        },
      });
      expect(wrapper.element).toMatchSnapshot();
    });

    it(':disabled', () => {
      const fn = vi.fn();
      const wrapper = mount({
        render() {
          return <Tag disabled onClick={fn}></Tag>;
        },
      });
      expect(fn).not.toHaveBeenCalled();
      expect(wrapper.element).toMatchSnapshot();
    });

    it(':variant:dark;theme:default', () => {
      const wrapper = mount({
        render() {
          return <Tag variant="dark" theme={'default'}></Tag>;
        },
      });
      expect(wrapper.element).toMatchSnapshot();
    });
    it(':variant:dark;theme:primary', () => {
      const wrapper = mount({
        render() {
          return <Tag variant="dark" theme={'primary'}></Tag>;
        },
      });
      expect(wrapper.element).toMatchSnapshot();
    });
    it(':variant:dark;theme:warning', () => {
      const wrapper = mount({
        render() {
          return <Tag variant="dark" theme={'warning'}></Tag>;
        },
      });
      expect(wrapper.element).toMatchSnapshot();
    });
    it(':variant:dark;theme:danger', () => {
      const wrapper = mount({
        render() {
          return <Tag variant="dark" theme={'danger'}></Tag>;
        },
      });
      expect(wrapper.element).toMatchSnapshot();
    });
    it(':variant:dark;theme:success', () => {
      const wrapper = mount({
        render() {
          return <Tag variant="dark" theme={'success'}></Tag>;
        },
      });
      expect(wrapper.element).toMatchSnapshot();
    });

    it(':variant:light;theme:default', () => {
      const wrapper = mount({
        render() {
          return <Tag variant="light" theme={'default'}></Tag>;
        },
      });
      expect(wrapper.element).toMatchSnapshot();
    });
    it(':variant:light;theme:primary', () => {
      const wrapper = mount({
        render() {
          return <Tag variant="light" theme={'primary'}></Tag>;
        },
      });
      expect(wrapper.element).toMatchSnapshot();
    });
    it(':variant:light;theme:warning', () => {
      const wrapper = mount({
        render() {
          return <Tag variant="light" theme={'warning'}></Tag>;
        },
      });
      expect(wrapper.element).toMatchSnapshot();
    });
    it(':variant:light;theme:danger', () => {
      const wrapper = mount({
        render() {
          return <Tag variant="light" theme={'danger'}></Tag>;
        },
      });
      expect(wrapper.element).toMatchSnapshot();
    });
    it(':variant:light;theme:success', () => {
      const wrapper = mount({
        render() {
          return <Tag variant="light" theme={'success'}></Tag>;
        },
      });
      expect(wrapper.element).toMatchSnapshot();
    });
    it(':variant:light;theme:default', () => {
      const wrapper = mount({
        render() {
          return <Tag variant="light" theme={'default'}></Tag>;
        },
      });
      expect(wrapper.element).toMatchSnapshot();
    });
    it(':variant:light;theme:primary', () => {
      const wrapper = mount({
        render() {
          return <Tag variant="light" theme={'primary'}></Tag>;
        },
      });
      expect(wrapper.element).toMatchSnapshot();
    });
    it(':variant:light;theme:warning', () => {
      const wrapper = mount({
        render() {
          return <Tag variant="light" theme={'warning'}></Tag>;
        },
      });
      expect(wrapper.element).toMatchSnapshot();
    });
    it(':variant:light;theme:danger', () => {
      const wrapper = mount({
        render() {
          return <Tag variant="light" theme={'danger'}></Tag>;
        },
      });
      expect(wrapper.element).toMatchSnapshot();
    });
    it(':variant:light;theme:success', () => {
      const wrapper = mount({
        render() {
          return <Tag variant="light" theme={'success'}></Tag>;
        },
      });
      expect(wrapper.element).toMatchSnapshot();
    });

    it(':shape:round', () => {
      const wrapper = mount({
        render() {
          return <Tag shape={'round'}></Tag>;
        },
      });
      expect(wrapper.element).toMatchSnapshot();
    });
    it(':shape:square', () => {
      const wrapper = mount({
        render() {
          return <Tag shape={'square'}></Tag>;
        },
      });
      expect(wrapper.element).toMatchSnapshot();
    });
    it(':shape:mark', () => {
      const wrapper = mount({
        render() {
          return <Tag shape={'mark'}></Tag>;
        },
      });
      expect(wrapper.element).toMatchSnapshot();
    });

    it(':maxWidth', () => {
      const wrapper = mount({
        render() {
          return <Tag max-width={100}></Tag>;
        },
      });
      expect(wrapper.element).toMatchSnapshot();
    });
  });

  describe('CheckTag:props', () => {
    it(':checked', () => {
      const wrapper = mount({
        render() {
          return <CheckTag checked></CheckTag>;
        },
      });
      expect(wrapper.element).toMatchSnapshot();
    });
    it(':disabled', () => {
      const fn = vi.fn();
      const wrapper = mount({
        render() {
          return <CheckTag disabled onClick={fn}></CheckTag>;
        },
      });
      expect(fn).not.toHaveBeenCalled();
      expect(wrapper.element).toMatchSnapshot();
    });
  });

  describe('Tag:slot', () => {
    it('<icon>', () => {
      const wrapper = mount(Tag, {
        scopedSlots: {
          icon: '<div></div>',
        },
      });
      expect(wrapper.element).toMatchSnapshot();
    });
  });

  describe('@event: Tag', () => {
    it('Event passthrough: click', () => {
      const fn = vi.fn();
      const wrapper = mount({
        render() {
          return <Tag onClick={fn}>text</Tag>;
        },
      });
      wrapper.findComponent(Tag).trigger('click');
      expect(fn).toHaveBeenCalled();
    });
    it('Event passthrough: close', () => {
      const fn = vi.fn();
      const wrapper = mount({
        render() {
          return <Tag closable onClose={fn}></Tag>;
        },
      });
      expect(wrapper.element).toMatchSnapshot();
      wrapper.find('.t-icon-close').trigger('click');
      expect(fn).toHaveBeenCalled();
    });
  });

  describe('@event: CheckTag', () => {
    it('Event passthrough: click', () => {
      const fn = vi.fn();
      const wrapper = mount({
        render() {
          return (
            <CheckTag checked onClick={fn}>
              text
            </CheckTag>
          );
        },
      });
      wrapper.findComponent(CheckTag).trigger('click');
      expect(fn).toBeCalled();
    });
    it('Event passthrough: trigger onClick props', () => {
      const fn = vi.fn();
      const wrapper = mount({
        render(h) {
          return h(CheckTag, {
            props: {
              onClick: fn,
            },
          });
        },
      }).findComponent(CheckTag);
      wrapper.findComponent(CheckTag).trigger('click');
      expect(fn).toBeCalled();
    });
  });
});
