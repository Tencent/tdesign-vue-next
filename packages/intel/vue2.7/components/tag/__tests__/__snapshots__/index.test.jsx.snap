// Vitest Snapshot v1

exports[`Tag or CheckTag > :Tag:props > :checked 1`] = `
<div
  checked="checked"
  class="t-tag t-tag--default t-tag--dark"
>
  <span />
</div>
`;

exports[`Tag or CheckTag > :Tag:props > :closable 1`] = `
<div
  class="t-tag t-tag--default t-tag--dark t-tag--close"
>
  <span />
  <svg
    class="t-icon t-icon-close t-tag__icon-close"
    fill="none"
    height="1em"
    viewBox="0 0 24 24"
    width="1em"
  >
    <path
      d="M7.05 5.64L12 10.59l4.95-4.95 1.41 1.41L13.41 12l4.95 4.95-1.41 1.41L12 13.41l-4.95 4.95-1.41-1.41L10.59 12 5.64 7.05l1.41-1.41z"
      fill="currentColor"
    />
  </svg>
</div>
`;

exports[`Tag or CheckTag > :Tag:props > :disabled 1`] = `
<div
  class="t-tag t-tag--default t-tag--dark t-is-disabled t-tag--disabled"
>
  <span />
</div>
`;

exports[`Tag or CheckTag > :Tag:props > :maxWidth 1`] = `
<div
  class="t-tag t-tag--default t-tag--dark t-tag--ellipsis"
>
  <span
    class="t-tag--text"
    style="max-width: 100px;"
  />
</div>
`;

exports[`Tag or CheckTag > :Tag:props > :shape:mark 1`] = `
<div
  class="t-tag t-tag--default t-tag--dark t-tag--mark"
>
  <span />
</div>
`;

exports[`Tag or CheckTag > :Tag:props > :shape:round 1`] = `
<div
  class="t-tag t-tag--default t-tag--dark t-tag--round"
>
  <span />
</div>
`;

exports[`Tag or CheckTag > :Tag:props > :shape:square 1`] = `
<div
  class="t-tag t-tag--default t-tag--dark"
>
  <span />
</div>
`;

exports[`Tag or CheckTag > :Tag:props > :size:large 1`] = `
<div
  class="t-tag t-tag--default t-tag--dark t-size-l"
>
  <span />
</div>
`;

exports[`Tag or CheckTag > :Tag:props > :size:medium 1`] = `<!---->`;

exports[`Tag or CheckTag > :Tag:props > :size:small 1`] = `<!---->`;

exports[`Tag or CheckTag > :Tag:props > :theme: 1`] = `
<div
  class="t-tag t-tag--default t-tag--dark"
>
  <span />
</div>
`;

exports[`Tag or CheckTag > :Tag:props > :theme:danger 1`] = `
<div
  class="t-tag t-tag--danger t-tag--dark"
>
  <span />
</div>
`;

exports[`Tag or CheckTag > :Tag:props > :theme:default 1`] = `
<div
  class="t-tag t-tag--default t-tag--dark"
>
  <span />
</div>
`;

exports[`Tag or CheckTag > :Tag:props > :theme:primary 1`] = `
<div
  class="t-tag t-tag--primary t-tag--dark"
>
  <span />
</div>
`;

exports[`Tag or CheckTag > :Tag:props > :theme:success 1`] = `
<div
  class="t-tag t-tag--success t-tag--dark"
>
  <span />
</div>
`;

exports[`Tag or CheckTag > :Tag:props > :theme:warning 1`] = `
<div
  class="t-tag t-tag--warning t-tag--dark"
>
  <span />
</div>
`;

exports[`Tag or CheckTag > :Tag:props > :variant:dark;theme:danger 1`] = `
<div
  class="t-tag t-tag--danger t-tag--dark"
>
  <span />
</div>
`;

exports[`Tag or CheckTag > :Tag:props > :variant:dark;theme:default 1`] = `
<div
  class="t-tag t-tag--default t-tag--dark"
>
  <span />
</div>
`;

exports[`Tag or CheckTag > :Tag:props > :variant:dark;theme:primary 1`] = `
<div
  class="t-tag t-tag--primary t-tag--dark"
>
  <span />
</div>
`;

exports[`Tag or CheckTag > :Tag:props > :variant:dark;theme:success 1`] = `
<div
  class="t-tag t-tag--success t-tag--dark"
>
  <span />
</div>
`;

exports[`Tag or CheckTag > :Tag:props > :variant:dark;theme:warning 1`] = `
<div
  class="t-tag t-tag--warning t-tag--dark"
>
  <span />
</div>
`;

exports[`Tag or CheckTag > :Tag:props > :variant:light;theme:danger 1`] = `
<div
  class="t-tag t-tag--danger t-tag--light"
>
  <span />
</div>
`;

exports[`Tag or CheckTag > :Tag:props > :variant:light;theme:danger 2`] = `
<div
  class="t-tag t-tag--danger t-tag--light"
>
  <span />
</div>
`;

exports[`Tag or CheckTag > :Tag:props > :variant:light;theme:default 1`] = `
<div
  class="t-tag t-tag--default t-tag--light"
>
  <span />
</div>
`;

exports[`Tag or CheckTag > :Tag:props > :variant:light;theme:default 2`] = `
<div
  class="t-tag t-tag--default t-tag--light"
>
  <span />
</div>
`;

exports[`Tag or CheckTag > :Tag:props > :variant:light;theme:primary 1`] = `
<div
  class="t-tag t-tag--primary t-tag--light"
>
  <span />
</div>
`;

exports[`Tag or CheckTag > :Tag:props > :variant:light;theme:primary 2`] = `
<div
  class="t-tag t-tag--primary t-tag--light"
>
  <span />
</div>
`;

exports[`Tag or CheckTag > :Tag:props > :variant:light;theme:success 1`] = `
<div
  class="t-tag t-tag--success t-tag--light"
>
  <span />
</div>
`;

exports[`Tag or CheckTag > :Tag:props > :variant:light;theme:success 2`] = `
<div
  class="t-tag t-tag--success t-tag--light"
>
  <span />
</div>
`;

exports[`Tag or CheckTag > :Tag:props > :variant:light;theme:warning 1`] = `
<div
  class="t-tag t-tag--warning t-tag--light"
>
  <span />
</div>
`;

exports[`Tag or CheckTag > :Tag:props > :variant:light;theme:warning 2`] = `
<div
  class="t-tag t-tag--warning t-tag--light"
>
  <span />
</div>
`;

exports[`Tag or CheckTag > @event: Tag > Event passthrough: close 1`] = `
<div
  class="t-tag t-tag--default t-tag--dark t-tag--close"
>
  <span />
  <svg
    class="t-icon t-icon-close t-tag__icon-close"
    fill="none"
    height="1em"
    viewBox="0 0 24 24"
    width="1em"
  >
    <path
      d="M7.05 5.64L12 10.59l4.95-4.95 1.41 1.41L13.41 12l4.95 4.95-1.41 1.41L12 13.41l-4.95 4.95-1.41-1.41L10.59 12 5.64 7.05l1.41-1.41z"
      fill="currentColor"
    />
  </svg>
</div>
`;

exports[`Tag or CheckTag > CheckTag:props > :checked 1`] = `
<div
  class="t-tag t-tag--primary t-tag--dark t-tag t-tag--check t-size-m t-tag--checked"
  tabindex="0"
>
  <span />
</div>
`;

exports[`Tag or CheckTag > CheckTag:props > :disabled 1`] = `
<div
  class="t-tag t-tag--default t-tag--dark t-is-disabled t-tag--disabled t-tag t-tag--check t-size-m t-tag--disabled"
>
  <span />
</div>
`;

exports[`Tag or CheckTag > Tag:slot > <icon> 1`] = `
<div
  class="t-tag t-tag--default t-tag--dark"
>
  <div />
  <span />
</div>
`;
